reporter/src/main/kotlin/model/EvaluatedModel.kt

```diff
@@ -19,7 +19,284 @@
+    val paths: List<EvaluatedPackagePath>,
+    val dependencyTrees: List<DependencyTreeNode>,
+    val violationResolutions: List<RuleViolationResolution>,
+    val violations: List<EvaluatedRuleViolation>,
+    val declaredLicenseStats: SortedMap<String, Int>,
+    val detectedLicenseStats: SortedMap<String, Int>,
+    val statistics: Statistics,
+    val repositoryConfiguration: String,
+    val customData: CustomData
+) {
+    companion object {
+        private val INT_ID_TYPES = listOf(
+            CopyrightStatement::class.java,
+            EvaluatedOrtIssue::class.java,
+            EvaluatedPackage::class.java,
+            EvaluatedPackagePath::class.java,
+            EvaluatedRuleViolation::class.java,
+            EvaluatedScanResult::class.java,
+            IssueResolution::class.java,
+            LicenseId::class.java,
+            PathExclude::class.java,
+            RuleViolationResolution::class.java,
+            ScopeName::class.java,
+            ScopeExclude::class.java
+        )
+
+        private val MAPPER_CONFIG: ObjectMapper.() -> Unit = {
+            registerKotlinModule()
+
+            registerModule(JavaTimeModule())
+            registerModule(IntIdModule(INT_ID_TYPES))
+
+            disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+
+            propertyNamingStrategy = PROPERTY_NAMING_STRATEGY
+        }
+
+        private val JSON_MAPPER = JsonMapper().apply { MAPPER_CONFIG() }
+
+        fun create(input: ReporterInput): EvaluatedModel = EvaluatedModelBuilder(input).build()
+    }
+
+    fun toJson(): String {
+        val tree = JSON_MAPPER.valueToTree<ObjectNode>(this)
+        tree.forEach { node ->
+            if (node is ArrayNode) {
+                if (!node.isEmpty && node[0].has("_id")) {
+                    val sortedChildren =
+                        node.elements().asSequence().toList().sortedBy { it["_id"].intValue() }.toList()
+                    node.removeAll()
+                    node.addAll(sortedChildren)
+                }
+            }
+        }
+        return tree.toPrettyString()
+    }
+}
+
+@JsonIgnoreProperties(value = ["paths"], allowGetters = true)
+data class EvaluatedPackage(
+    val id: Identifier,
+    val isProject: Boolean,
+    val definitionFilePath: String,
+    val purl: String = id.toPurl(),
+    val declaredLicenses: List<LicenseId>,
+    val declaredLicensesProcessed: EvaluatedProcessedDeclaredLicense,
+    val detectedLicenses: Set<LicenseId>,
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val concludedLicense: SpdxExpression? = null,
+    val description: String,
+    val homepageUrl: String,
+    val binaryArtifact: RemoteArtifact,
+    val sourceArtifact: RemoteArtifact,
+    val vcs: VcsInfo,
+    val vcsProcessed: VcsInfo = vcs.normalize(),
+    val curations: List<PackageCurationResult>,
+    @JsonIdentityReference(alwaysAsId = true)
+    var paths: MutableList<EvaluatedPackagePath> = mutableListOf(),
```

Comment:
Why is it mutable? ..Applies to `var` and to `MutableList`.

Reply:
It's a leftover from a deserialization hack, but since I dropped deserialization support I'll make it a `val` again.